zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/GroupRemoveEntityProcessor.java

```diff
@@ -0,0 +1,124 @@
+import io.camunda.zeebe.stream.api.state.KeyGenerator;
+
+public class GroupRemoveEntityProcessor implements DistributedTypedRecordProcessor<GroupRecord> {
+
+  private final GroupState groupState;
+  private final UserState userState;
+  private final MappingState mappingState;
+  private final AuthorizationCheckBehavior authCheckBehavior;
+  private final KeyGenerator keyGenerator;
+  private final StateWriter stateWriter;
+  private final TypedRejectionWriter rejectionWriter;
+  private final TypedResponseWriter responseWriter;
+  private final CommandDistributionBehavior commandDistributionBehavior;
+
+  public GroupRemoveEntityProcessor(
+      final GroupState groupState,
+      final UserState userState,
+      final MappingState mappingState,
+      final AuthorizationCheckBehavior authCheckBehavior,
+      final KeyGenerator keyGenerator,
+      final Writers writers,
+      final CommandDistributionBehavior commandDistributionBehavior) {
+    this.groupState = groupState;
+    this.userState = userState;
+    this.mappingState = mappingState;
+    this.authCheckBehavior = authCheckBehavior;
+    this.keyGenerator = keyGenerator;
+    stateWriter = writers.state();
+    rejectionWriter = writers.rejection();
+    responseWriter = writers.response();
+    this.commandDistributionBehavior = commandDistributionBehavior;
+  }
+
+  @Override
+  public void processNewCommand(final TypedRecord<GroupRecord> command) {
+    final var record = command.getValue();
+    final var groupKey = record.getGroupKey();
+    final var persistedRecord = groupState.get(groupKey);
+    if (persistedRecord.isEmpty()) {
+      final var errorMessage =
+          "Expected to update group with key '%s', but a group with this key does not exist."
+              .formatted(groupKey);
+      rejectionWriter.appendRejection(command, RejectionType.NOT_FOUND, errorMessage);
+      responseWriter.writeRejectionOnCommand(command, RejectionType.NOT_FOUND, errorMessage);
+      return;
+    }
+
+    final var authorizationRequest =
+        new AuthorizationRequest(command, AuthorizationResourceType.GROUP, PermissionType.UPDATE)
+            .addResourceId(persistedRecord.get().getName());
+    if (!authCheckBehavior.isAuthorized(authorizationRequest)) {
+      final var errorMessage =
+          UNAUTHORIZED_ERROR_MESSAGE.formatted(
+              authorizationRequest.getPermissionType(), authorizationRequest.getResourceType());
+      rejectionWriter.appendRejection(command, RejectionType.UNAUTHORIZED, errorMessage);
+      responseWriter.writeRejectionOnCommand(command, RejectionType.UNAUTHORIZED, errorMessage);
+      return;
+    }
+
+    final var entityKey = record.getEntityKey();
+    final var entityType = record.getEntityType();
+    if (!isEntityPresent(entityKey, entityType)) {
+      final var errorMessage =
+          "Expected to remove an entity with key '%s' and type '%s' from group with key '%s', but the entity doesn't exist."
```

Comment:
🔧 This should be aligned with above error message ".... but the entity does not exist."